packages/asu-core/__tests__/aws/services/cache.service.spec.ts

```diff
@@ -0,0 +1,149 @@
+            const expectedKey = 'key';
+            const expectedValue = 'value';
+
+            cacheClient.get.mockResolvedValue(expectedValue);
+
+            // Act
+            const result = await service.getValue(expectedKey, async () => {
+                return null;
+            });
+
+            // Assert
+            expect(result).toBe(expectedValue);
+        });
+
+        it('should call given callback when cache fails to connect', async () => {
+            // Arrange
+            const { service, cacheClient } = setup();
+            const expectedKey = 'key';
+            const expectedValue = 'value';
+
+            cacheClient.connect.mockImplementation(() => {
+                throw 'connection timed out';
+            });
+
+            // Act
+            const result = await service.getValue(expectedKey, async () => {
+                return expectedValue;
+            });
+
+            // Assert
+            expect(result).toBe(expectedValue);
+        });
+
+        it('should call given callback when cache get returns null', async () => {
+            // Arrange
+            const { service, cacheClient } = setup();
+            const expectedKey = 'key';
+            const expectedValue = 'value';
+
+            cacheClient.get.mockResolvedValue(null);
+
+            // Act
+            const result = await service.getValue(expectedKey, async () => {
+                return expectedValue;
+            });
+
+            // Assert
+            expect(result).toBe(expectedValue);
+        });
+
+        it('should call given callback when cache get throws exception', async () => {
+            // Arrange
+            const { service, cacheClient } = setup();
+            const expectedKey = 'key';
+            const expectedValue = 'value';
+
+            cacheClient.get.mockImplementation(() => {
+                throw 'Failed to get value';
+            });
+
+            // Act
+            const result = await service.getValue(expectedKey, async () => {
+                return expectedValue;
+            });
+
+            // Assert
+            expect(result).toBe(expectedValue);
+        });
+
+        it('should set value if key was not previously cached', async () => {
+            // Arrange
+            const { service, cacheClient } = setup();
+            const expectedKey = 'key';
+            const expectedValue = 'value';
+
+            cacheClient.get.mockResolvedValue(null);
+
+            // Act
+            const result = await service.getValue(expectedKey, async () => {
+                return expectedValue;
+            });
+
+            // Assert
+            expect(result).toBe(expectedValue);
+            expect(cacheClient.set.mock.calls[0][0]).toBe(expectedKey);
+            expect(cacheClient.set.mock.calls[0][1]).toBe(expectedValue);
+            expect(cacheClient.set.mock.calls[0][2]).toBe('EX');
+            expect(cacheClient.set.mock.calls[0][3]).toBe(TimeToLive.OneHour);
```

Comment:
Is there any benefit to checking params this way versus condensing it to this?: 
```
expect(cacheClient.set).toHaveBeenCalledWith(expectedKey, expectedValue, 'EX', TimeToLive.OneHour);
```

Reply:
Nope, I like your pattern better! Thank you